### PR TITLE
[PL-24911]: Make project name as optional param in Azure Repo APIs

### PR DESCRIPTION
**Pull request by DeepakPatankar**
For the list Repo api of the Azure repo. the project name is not a required parameter.In Harness, we support an Account type connectors which won’t contain the project name. When we are doing the test connection of this connector, we call the scm API to list all the repositories of an organization.  Currently the scm code doesn’t allow the project name as null, which I fixed in this ticket

### CODE REVIEW

File: scm/driver/azure/azure.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func New(uri, owner, project string) (*scm.Client, error) {
 	if !strings.HasSuffix(base.Path, "/") {
 		base.Path = base.Path + "/"
 	}
-	if owner == "" || project == "" {
+	if owner == "" {
 		return nil, fmt.Errorf("azure owner and azure project are required")
```
Comment (mohitg0795) [inline scm/driver/azure/azure.go:29]: logline should be fixed accordingly.

**Review by tphoney — APPROVED** [review]

**Comment by tphoney** [comment]
@DeepakPatankar looks good with the change, but you will need to fix the unit tests first.

File: scm/driver/azure/repo.go [hunk 2]
```diff
@@ -40,7 +40,12 @@ func (s *RepositoryService) FindPerms(ctx context.Context, repo string) (*scm.Pe
 // List returns the user repository list.
 func (s *RepositoryService) List(ctx context.Context, opts scm.ListOptions) ([]*scm.Repository, *scm.Response, error) {
 	// https://docs.microsoft.com/en-us/rest/api/azure/devops/git/repositories/list?view=azure-devops-rest-6.0
-	endpoint := fmt.Sprintf("%s/%s/_apis/git/repositories?api-version=6.0", s.client.owner, s.client.project)
+	var endpoint string
+	if s.client.project != "" {
```
Comment (allofthesepeople) [inline scm/driver/azure/repo.go:44]: nit: prefer positive to negative conditionals as they're easier/quicker to understand.

**Review by allofthesepeople — APPROVED** [review]
minor nit on conditional

File: scm/driver/azure/repo.go (hunk 2 above)
Comment (DeepakPatankar) [reply scm/driver/azure/repo.go]: done

File: scm/driver/azure/azure.go (hunk 1 above)
Comment (DeepakPatankar) [reply scm/driver/azure/azure.go]: done

**Review by tphoney — COMMENTED** [review]
If a user creates a scm client and does not set the project, it will fail spectacularly for all calls except for List repositories. It does not feel like there is any protection against this. with the change to 
```
if owner == "" || project == "" {
		return nil, fmt.Errorf("azure owner and azure project are required")
	}
```

**Pull request merged**